lib/src/layer/marker_layer.dart

```diff
@@ -102,49 +103,51 @@ class MarkerLayer extends StatelessWidget {
   Widget build(BuildContext context) {
     final map = MapCamera.of(context);
 
-    return Stack(
-      children: (List<Widget?>.generate(
-        markers.length,
-        (i) {
-          final m = markers[i];
-
-          // Resolve real alignment
-          final left = 0.5 * m.width * ((m.alignment ?? alignment).x + 1);
-          final top = 0.5 * m.height * ((m.alignment ?? alignment).y + 1);
-          final right = m.width - left;
-          final bottom = m.height - top;
-
-          // Perform projection
-          final pxPoint = map.project(m.point);
-
-          // Cull if out of bounds
-          if (!map.pixelBounds.containsPartialBounds(
-            Bounds(
-              Point(pxPoint.x + left, pxPoint.y - bottom),
-              Point(pxPoint.x - right, pxPoint.y + top),
-            ),
-          )) return null;
-
-          // Apply map camera to marker position
-          final pos = pxPoint.subtract(map.pixelOrigin);
-
-          return Positioned(
-            key: m.key,
-            width: m.width,
-            height: m.height,
-            left: pos.x - right,
-            top: pos.y - bottom,
-            child: (m.rotate ?? rotate)
-                ? Transform.rotate(
-                    angle: -map.rotationRad,
-                    alignment: (m.alignment ?? alignment) * -1,
-                    child: m.child,
-                  )
-                : m.child,
-          );
-        },
-      )..retainWhere((w) => w != null))
-          .cast(),
+    return MobileLayerTransformer(
+      child: Stack(
+        children: (List.generate(
```

Comment:
It would probably be better to use a callback:
```dart
Iterable<Widget> createMarkers(List<Marker> markers) sync* {
  // ...
}
```

You'd be able to use yield to add a valid marker instead of returning null then doing a `retainWhere` and a `cast` (hence avoiding to iterate multiple times over your marker list).

```dart
Stack(
  children: [ ...createMarkers(markers) ],
),
```

Reply:
Added in e991671. Just to note, this was introduced in #1659, not this PR.